modules/core/src/main/scala/compiler.scala

```diff
@@ -173,7 +173,7 @@ object QueryMinimizer {
 
   def minimizeText(text: String): Either[String, String] = {
     for {
-      doc <- GraphQLParser.Document.parseOnly(text).either
+      doc <- GraphQLParser.Document.parseAll(text).leftMap(_.expected.toList.mkString(","))
```

Comment:
while this may be okay for the short term, I would probably make a custom `Parser.Error => String` function can call that (presumably also with access to `text` and possibly using `cats.parse.LocationMap`).